Documentation/Porting.md

```diff
@@ -0,0 +1,83 @@
+# Porting to new Operating System Environments
+
+This document outlines the steps required to port Ladybird to a new OS environment, and expectations for such ports to
+be accepted into the mainline repository.
+
+## Types of Ports
+
+There are two types of ports that can be made to Ladybird:
+
+- Chrome port: We define the "browser chrome" as the UI layer. This includes the browser window, tabs, address bar, etc.
+- Platform port: This includes the underlying platform-specific code that interacts with the OS. This includes things like
+  file I/O, networking, and process management.
+
+### Chrome Ports
+
+There are currently three supported chrome ports:
+
+- Qt6: The generic chrome port.
+- AppKit/Cocoa: The macOS native port, which uses the AppKit framework.
+- Headless: A headless port that does not have a UI, used for testing.
+
+### Platform Ports
+
+There are currently two supported platform ports:
+
+- GNU/Linux: The Linux platform port that may work on other POSIX platforms
+- macOS: The macOS platform port
+
+Many other POSIX desktop platforms are known to work, or have worked in the past, but are not officially supported.
+These include Alpine Linux, FreeBSD, OpenBSD, and Haiku. Support for these Unixen is community-supported, meaning there
+is no regular pre-commit CI guaranteeing that each master commit keeps them working. Patches to bring them back in line
```

Comment:
`pre-commit` does not seem necessary here. And it kinda implies that we have pre-commit tests for the binary on other platforms.

Reply:
True, we don't use the term `pre-commit` anywhere else in the project. I was thinking of PR CI as pre-commit, since it's before it goes into master, but I've removed the qualifier from CI for consistency.